pkg/controller/node/node_controller.go

```diff
@@ -995,3 +1011,74 @@ func getErrorString(err error) string {
 	}
 	return ""
 }
+
+// alertIfImportantFilesPending checks to see if there are any important files in the
+// machineconfig that the pool should be moving to, and alerts if there are
+func (ctrl *Controller) alertIfImportantFilesPending(pool *mcfgv1.MachineConfigPool) {
+
+	// Figure out what files differ between pool.Spec and pool.Status
+	diff, err := ctrl.diffMachineConfigFilesForPool(pool)
+	if err != nil {
+		glog.Warningf("Error getting %s pool file diff: %v", pool.Name, err)
+		return
+	}
+	for _, path := range diff {
+
+		// If it's the kubelet CA, then alert
+		if path == kubeletCAFilePath {
+			glog.V(2).Infof("Kubelet CA is stuck in paused pool. Firing alert.")
+			ctrlcommon.MachineConfigControllerPausedPoolKubeletCA.WithLabelValues(pool.Name).Set(1)
+		}
+	}
+
+}
+
+// resetImportantFileAlerts turns off any "paused file" alerts that were firing for the pool
+func (ctrl *Controller) resetImportantFileAlerts(pool *mcfgv1.MachineConfigPool) {
+
+	// Set the alert for this pool back to zero
+	ctrlcommon.MachineConfigControllerPausedPoolKubeletCA.WithLabelValues(pool.Name).Set(0)
+
+}
+
+// diffMachineConfigFilesForPool retrieves the machine config specified by the pool spec and status,
+// compares them, and returns the list of file paths that are different
+func (ctrl *Controller) diffMachineConfigFilesForPool(pool *mcfgv1.MachineConfigPool) ([]string, error) {
+
+	// The config we would be going to
+	specName := pool.Spec.Configuration.Name
+	// The config we're in right now
+	statusName := pool.Status.Configuration.Name
```

Comment:
A convention I've seen elsewhere in the MCO is to refer to these as `currentConfig` (the config we're currently in) and `pendingConfig` (the config we're moving to).

Reply:
I've updated the language to favor the current/pending convention for clarity - my head was full of pool specs and statuses from hanging out in the node controller.